WordPressKit/RemoteVideoPressVideo.swift

```diff
@@ -0,0 +1,128 @@
+import Foundation
+
+/// This enum matches the privacy setting constants defined in Jetpack:
+/// https://github.com/Automattic/jetpack/blob/a2ccfb7978184e306211292a66ed49dcf38a517f/projects/packages/videopress/src/utility-functions.php#L13-L17
+@objc public enum VideoPressPrivacySetting: Int, Encodable {
+    case isPublic = 0
+    case isPrivate = 1
+    case siteDefault = 2
+}
+
+@objcMembers public class RemoteVideoPressVideo: NSObject, Encodable {
+
+    /// The following properties match the response parameters from the `videos` endpoint:
+    /// https://developer.wordpress.com/docs/api/1.1/get/videos/%24guid/
+    ///
+    /// However, it's missing the following parameters that could be added in the future if needed:
+    /// - files
+    /// - file_url_base
+    /// - upload_date
+    /// - files_status
+    /// - subtitles
+    public var id: String
+    public var title: String?
+    public var videoDescription: String?
+    public var width: Int?
+    public var height: Int?
+    public var duration: Int?
+    public var displayEmbed: Bool?
+    public var allowDownload: Bool?
+    public var rating: String?
+    public var privacySetting: VideoPressPrivacySetting = .siteDefault
+    public var posterURL: URL?
+    public var originalURL: URL?
+    public var watermarkURL: URL?
+    public var bgColor: String?
+    public var blogId: Int?
+    public var postId: Int?
+    public var finished: Bool?
+
+    public var token: String?
+
+    enum CodingKeys: String, CodingKey {
+        case id, title, description, width, height, duration, displayEmbed, allowDownload, rating, privacySetting, posterURL, originalURL, watermarkURL, bgColor, blogId, postId, finished, token
```

Comment:
We may be able to add `videoDescription = "description"` to allow auto synthesis of `Encodable`.

Reply:
Yep, applied in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/581/commits/b901dc8f0ae251f2698a1bfa2e7c2a1c4599e6c1.